Cilsil/Utils/ProgramState.cs

```diff
@@ -392,28 +385,38 @@ public void PushRetExpr()
                     PreviousNode = node ?? PreviousNode,
                     PreviousStack = ProgramStack.Clone(),
                     NextAvailableTemporaryVariableId = NextAvailableTemporaryVariableId,
-                    PreviousReturnedExpression = PreviousReturnedExpression,
                     PreviousReturnedType = PreviousReturnedType,
                 });
 
         /// <summary>
         /// Pops an instruction to be parsed.
         /// </summary>
         /// <returns>The instruction to be parsed and its previous node.</returns>
-        public (Instruction, CfgNode) PopInstruction(bool popProgramStack = true)
+        public (Instruction, CfgNode) PopInstruction()
         {
             var snapshot = InstructionsStack.Pop();
             PreviousNode = snapshot.PreviousNode;
             CurrentInstruction = snapshot.Instruction;
-            if (popProgramStack)
-                ProgramStack = snapshot.PreviousStack;
+            ProgramStack = snapshot.PreviousStack;
             NextAvailableTemporaryVariableId = snapshot.NextAvailableTemporaryVariableId;
 
             var currentSequencePoint =
                 Method.DebugInformation.GetSequencePoint(CurrentInstruction);
             if (currentSequencePoint != null)
             {
                 var newLocation = Location.FromSequencePoint(currentSequencePoint);
+                var previousInstruction = CurrentInstruction.Previous;
+                while (newLocation.Line - CurrentLocation.Line >= 100 && previousInstruction != null)
```

Comment:
why this condition?

Reply:
It was for fixing the huge line problem we saw in CFG

Reply:
> why this condition?

It was for fixing the huge line problem we saw in CFG